src/strategies/GmStrategy.sol

```diff
@@ -0,0 +1,242 @@
+
+    bytes32 public constant EXECUTE_DEPOSIT_FEATURE_DISABLED = keccak256(abi.encode("EXECUTE_DEPOSIT_FEATURE_DISABLED"));
+
+    IGmxV2ExchangeRouter public immutable GMX_ROUTER;
+    IGmxReader public immutable GMX_READER;
+    IGmxDataStore public immutable DATASTORE;
+    address public immutable DEPOSIT_VAULT;
+    address public immutable SYNTHETICS_ROUTER;
+    address public immutable REFUND_TO;
+    address public immutable LONG_TOKEN;
+    IMultiRewardsStaking public immutable STAKING;
+
+    address public exchange;
+    uint256 public callbackGasLimit = 2_000_000;
+    bytes32 public orderKey;
+
+    /// @dev Keep in memory the max balance once the GMX tokens are deposited
+    uint256 private maxBalance;
+
+    modifier onlyDepositHandler() {
+        if (msg.sender != address(GMX_ROUTER.depositHandler())) {
+            revert ErrUnauthorized();
+        }
+        _;
+    }
+
+    receive() external payable virtual {
+        (bool success, ) = REFUND_TO.call{value: msg.value}("");
+
+        // ignore failures
+        if (!success) {
+            return;
+        }
+    }
+
+    constructor(
+        address _strategyToken,
+        IBentoBoxV1 _degenBox,
+        IGmxV2ExchangeRouter _gmxRouter,
+        IGmxReader _gmxReader,
+        address _syntheticsRouter,
+        address _refundTo,
+        address _staking
+    ) BaseStrategy(IERC20(_strategyToken), _degenBox) {
+        assert(IMultiRewardsStaking(_staking).stakingToken() == _strategyToken);
+
+        feeBips = 200; // 2%
+        feeCollector = msg.sender;
+
+        GMX_ROUTER = _gmxRouter;
+        GMX_READER = _gmxReader;
+        SYNTHETICS_ROUTER = _syntheticsRouter;
+        DATASTORE = IGmxDataStore(_gmxRouter.dataStore());
+        DEPOSIT_VAULT = IGmxV2DepositHandler(_gmxRouter.depositHandler()).depositVault();
+        REFUND_TO = _refundTo;
+
+        IGmxV2Market.Props memory props = GMX_READER.getMarket(address(DATASTORE), _strategyToken);
+        LONG_TOKEN = props.longToken;
+
+        STAKING = IMultiRewardsStaking(_staking);
+
+        _strategyToken.safeApprove(address(_staking), type(uint256).max);
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////////////////
+    /// EXECUTORS
+    //////////////////////////////////////////////////////////////////////////////////////////////
+
+    /// @param _rewardToken Reward token from the staking contract
+    /// @param _marketInputToken Same as _rewardToken when _swapData is empty,
+    /// otherwise the token to use as market token input
+    /// @param _amount Amount of reward tokens to swap
+    function run(
+        address _rewardToken,
+        address _marketInputToken,
+        uint256 _amount,
+        uint256 _marketMinOut,
+        uint256 _executionFee,
+        bytes memory _swapData,
+        uint256 _maxBentoBoxAmountIncreaseInBips,
+        uint256 _maxBentoBoxChangeAmountInBips
+    ) external payable onlyExecutor {
```

Comment:
Can we run a gelato task that needs funds?
Maybe we need to have some ETH on this contract similar to the old mSpellSender contracts

Reply:
We can have ETH in the gelatoProxy, we are doing this aws